crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -742,12 +787,19 @@ pub struct RpcServer {
     /// The address of the ws server
     ws_local_addr: Option<SocketAddr>,
     /// http server
-    http: Option<Server>,
+    http: Option<HttpServer>,
     /// ws server
     ws: Option<Server>,
     /// ipc server
     ipc: Option<IpcServer>,
 }
+/// Http Servers Enum
+pub enum HttpServer {
+    /// Http server
+    Plain(Server),
+    /// Http server with cors
+    WithCors(Server<Stack<CorsLayer, Identity>>),
+}
```

Comment:
this is okay for now, perhaps we find a better solution but for now this is ok

Reply:
We could dyn dispatch